pyproject.toml

```diff
@@ -16,6 +16,9 @@ urls.Homepage = "https://github.com/tox-dev/pipdeptree"
 urls.Source = "https://github.com/tox-dev/pipdeptree"
 urls.Tracker = "https://github.com/tox-dev/pipdeptree/issues"
 requires-python = ">=3.7"
+dependencies = [
+  "importlib_metadata; python_version < '3.8'"
```

Comment:
We cannot take on any dependencies, as that would mean the `--python` flag would fail if the target python doesn't have the given dependency installed.

Reply:
It will not really fail because we have `except ImportError: return default`, it will just show the version as `?`.

And the target venv probably has it installed anyway as a transitive dependency, too many packages depend on it (like pytest).

We can still remove the dependency, but then what to do for 3.7?
I can think of few ways to handle it:
1) Current behavior (in this branch): Just try importing `importlib_metadata` but without adding it as a dependency and fallback to default `?` version if it can't be imported.
2) Fall back to the `package.__version__` heuristic that is implemented in the main branch, with some special casing for `setuptools` to avoid the `AssertionError`.
3) Report `?` unconditionally
4) Drop support for python 3.7

Reply:
We need to do 2 realistically.

Reply:
Okay so the messy one that I was trying to avoid :)

I had to force-push to handle the merge conflicts anyway, so I started from scratch, and I think I have a cleaner way now.

The `package.__version__` way is worse than using `importlib_metadata` so I am still trying `importlib_metadata` first, but if that fails then the old logic runs (except in case of setuptools, where we just return `?`).